rs/bitcoin/kyt/src/state.rs

```diff
@@ -128,52 +130,35 @@ impl Drop for FetchGuard {
     }
 }
 
-#[derive(Clone, Debug)]
+#[derive(CandidType, Clone, Debug, Deserialize, Serialize)]
 pub struct Config {
-    pub network: Network,
+    pub network: BtcNetwork,
 }
 
 impl Default for Config {
     fn default() -> Self {
         Self {
-            network: Network::Bitcoin,
+            network: BtcNetwork::Mainnet,
         }
     }
 }
 
-impl From<InitArg> for Config {
-    fn from(arg: InitArg) -> Self {
-        let network = match arg.network {
-            BtcNetwork::Mainnet => Network::Bitcoin,
-            BtcNetwork::Testnet => Network::Testnet,
-            BtcNetwork::Regtest => Network::Regtest,
-        };
-        Self { network }
-    }
-}
-
-impl From<&Config> for InitArg {
-    fn from(arg: &Config) -> Self {
-        let network = match arg.network {
-            Network::Bitcoin => BtcNetwork::Mainnet,
-            Network::Testnet => BtcNetwork::Testnet,
-            Network::Regtest => BtcNetwork::Regtest,
-            _ => panic!("unsupported network type: Signet"),
-        };
-        Self { network }
-    }
+#[derive(CandidType, Clone, Debug, Deserialize, Serialize)]
```

Comment:
1. `ConfigState` should not need to be a `CandidType`, it's an implementation detail not exposed by the canister
2. `Candid` is the worst by far in terms of serialization/deserialization performance (see those [benchmarks](https://github.com/dfinity/wasm-benchmarks/tree/main)) and it doesn't always play nice with `serde` annotations (not all are supported). I would advise not using `Candid` for serialization into stable memory. I think the easiest here would be to use `ciborum`. See for example the [LSO](https://github.com/dfinity/ic/blob/48f24e7e48590122130186e894d8096b12e9b772/rs/ethereum/ledger-suite-orchestrator/src/state/mod.rs#L499). WDYT?

Reply:
Thanks for sharing the concern on Candid. I think Candid should be fine to use when performance doesn't matter, like in this case for a simple config object. Introducing `ciborum` would be adding another dependency. WDYT?

Reply:
yes, that's a fair point. I guess I'm generally biased towards keeping Candid as much as possible outside of canister logic (and be used only for the external API) in favour of using well-established other solutions that just do deser. It's fine either way.

Reply:
After a second thought, I decided to switch to ciborium just in case more data needs to be persisted. 

BTW, just a fun fact, adding ciborium increases wasm binary size by about 16kB :-)